### DISCUSSION TITLE
Recent changes make Insomnia more difficult to use for GraphQL

### DISCUSSION THREAD

**Discussion by walker-tx** [Kong/insomnia#5595] [Feedback]
Howdy!

I've been a doting Insomnia user for some time now. I've bragged about it to coworkers and gotten friends to start using it, too. I've loved how easy it's been to use for GraphQL requests - esp since I can use a plugin to auth w/ Cognito. Insomnia updated recently though and it's made working GraphQL less smooth. I'm not sure if this is a product of my local config, but I can't find any preferences to switch things back to the way they were. For instance:

## Execute the Query that the cursor is in

Previously, to execute a gql request, I only needed to put my cursor w/in the operation that I wanted to execute. This was super handy because I often have a dozen or so different requests in the body. Now, I have to select the query using the dropdown next to "Schema". This is a large adjustment from the previous workflow - I find myself accidentally executing queries that I selected w/ my cursor, but forgot to select in the dropdown. I've accidentally sent some mutations that had to be undone :(. 

What's more - oftentimes the operation dropdown isn't selectable. I have to type something (anything) in the request body so it becomes enabled so I can select an operation to run. It's as if it's not validating on mount.

## Autocomplete shortcut isn't working

Previously, the autocomplete hotkeys (Ctrl+Space) worked for GraphQL also. But now Ctrl+Space only works in like the URL bar - for templates. For some reason, Shift+Space will bring up the autocomplete menu for GQL, but that also enters a space into the request body which has to be deleted. Well, I guess it doesn't have to, but 3-space tabs aren't common and this can cause the alignment to be all over the place.

## Click for documentation doesn't work

Before, I could do Cmd+Click on a resolver to pull up its documentation. That doesn't work before. I've tried Ctrl and Alt/Opt click, too, but nothing. Of course, I can just click the resolver name, and a tooltip will appear to a link w/in the GQL API documentation, so this gripe is less signification if I'm honest.

## Variables default to `null` now

Variables default to `null` now. GQL doesn't care about this, but Insomnia doesn't like it, and actually underlines the `null` it filled in with a red squiggly.

# Concluding

I just want to say that I like Insomnia very much, and all of these issues have been more recent (last couple of weeks). These seem to have cropped up after the sidebar with the "Home" button appeared. Is this an issue with my local configs? If not, would y'all please consider bringing back some of the old QOL features for GQL users?

Thank you!

**Comment by ChrispyKremeDonuts** [comment]
Is there a way to downgrade to the last version? I use insomnia solely to make graphql requests and now it is unusable for me.

**Reply by alvaro-s-oliveira** [reply]
They keep making it worse to use, it boggles my mind how is that possible. Now they removed the color coded requests on the sidebar, how could they come to a conclusion that this would be a a good change.

**Reply by robmellett** [reply]
Totally agree. What version did you downgrade to?

**Reply by ganchdev** [reply]
I don't use GraphQL much but downgraded to version `2022.5.1` on my Mac and I'm super happy.

Link: https://github.com/Kong/insomnia/releases/tag/core%402022.5.1
